internal/timecraft/process.go

```diff
@@ -82,12 +83,13 @@ const (
 )
 
 // NewProcessManager creates an ProcessManager.
-func NewProcessManager(ctx context.Context, registry *timemachine.Registry, runtime wazero.Runtime, serverFactory *ServerFactory) *ProcessManager {
+func NewProcessManager(ctx context.Context, registry *timemachine.Registry, runtime wazero.Runtime, serverFactory *ServerFactory, adapter func(ProcessID, wasi.System) wasi.System) *ProcessManager {
```

Comment:
I'm not really fond of this extra parameter, I'd suggest we revisit when we bring back the process API.